### ISSUE TITLE
Extension spawns orphan python.exe processes that aren't killed when VSCode/extension_host are shutdown

### ISSUE THREAD

**Issue by Kein** [microsoft/vscode-mypy#291]
## At a glance:
![image](https://github.com/microsoft/vscode-mypy/assets/623756/063b3948-fd21-4da8-914b-167f8fc521ce)

## In detail:
Env command line reports this:
```
c:\Programs\Python390\python.exe -m mypy.dmypy --status-file e:\Temp\.vscode.dmypy_status\status-72e28ba3-939e-4538-8ce8-0f67c32a71f7.json daemon --options-data=\"gASVGgoAAAAAAAB9lCiMCmJ1aWxkX3R5cGWUSwCMDnB5dGhvbl92ZXJzaW9ulEsDSwmGlIwRcHl0aG9uX2V4ZWN1dGFibGWUjCBjOlxQcm9ncmFtc1xQeXRob24zOTBccHl0aG9uLmV4ZZSMCHBsYXRmb3JtlIwFd2luMzKUjBRjdXN0b21fdHlwaW5nX21vZHVsZZROjBNjdXN0b21fdHlwZXNoZWRfZGlylE6MF2Fic19jdXN0b21fdHlwZXNoZWRfZGlylE6MCW15cHlfcGF0aJRdlIwLcmVwb3J0X2RpcnOUfZSMGG5vX3NpbGVuY2Vfc2l0ZV9wYWNrYWdlc5SJjBBub19zaXRlX3BhY2thZ2VzlImMFmlnbm9yZV9taXNzaW5nX2ltcG9ydHOUiYwhaWdub3JlX21pc3NpbmdfaW1wb3J0c19wZXJfbW9kdWxllImMDmZvbGxvd19pbXBvcnRzlIwGbm9ybWFslIwYZm9sbG93X2ltcG9ydHNfZm9yX3N0dWJzlImMEm5hbWVzcGFjZV9wYWNrYWdlc5SIjBZleHBsaWNpdF9wYWNrYWdlX2Jhc2VzlImMB2V4Y2x1ZGWUXZSMFWRpc2FsbG93X2FueV9nZW5lcmljc5SJjBdkaXNhbGxvd19hbnlfdW5pbXBvcnRlZJSJjBFkaXNhbGxvd19hbnlfZXhwcpSJjBZkaXNhbGxvd19hbnlfZGVjb3JhdGVklImMFWRpc2FsbG93X2FueV9leHBsaWNpdJSJjBZkaXNhbGxvd191bnR5cGVkX2NhbGxzlImMFXVudHlwZWRfY2FsbHNfZXhjbHVkZZRdlIwVZGlzYWxsb3dfdW50eXBlZF9kZWZzlImMGGRpc2FsbG93X2luY29tcGxldGVfZGVmc5SJjBJjaGVja191bnR5cGVkX2RlZnOUiYwbZGlzYWxsb3dfdW50eXBlZF9kZWNvcmF0b3JzlImMGGRpc2FsbG93X3N1YmNsYXNzaW5nX2FueZSJjBR3YXJuX2luY29tcGxldGVfc3R1YpSJjBR3YXJuX3JlZHVuZGFudF9jYXN0c5SJjA53YXJuX25vX3JldHVybpSIjA93YXJuX3JldHVybl9hbnmUiYwTd2Fybl91bnVzZWRfaWdub3Jlc5SJjBN3YXJuX3VudXNlZF9jb25maWdzlImMDWlnbm9yZV9lcnJvcnOUiYwPc3RyaWN0X29wdGlvbmFslIiMEnNob3dfZXJyb3JfY29udGV4dJSJjAxjb2xvcl9vdXRwdXSUiYwNZXJyb3Jfc3VtbWFyeZSJjBFpbXBsaWNpdF9vcHRpb25hbJSJjBFpbXBsaWNpdF9yZWV4cG9ydJSIjBVhbGxvd191bnR5cGVkX2dsb2JhbHOUiYwSYWxsb3dfcmVkZWZpbml0aW9ulImMD3N0cmljdF9lcXVhbGl0eZSJjBJzdHJpY3RfY29uY2F0ZW5hdGWUiYwMZXh0cmFfY2hlY2tzlImMEHdhcm5fdW5yZWFjaGFibGWUiYwLYWx3YXlzX3RydWWUXZSMDGFsd2F5c19mYWxzZZRdlIwSZGlzYWJsZV9lcnJvcl9jb2RllF2UjBRkaXNhYmxlZF9lcnJvcl9jb2Rlc5SPlIwRZW5hYmxlX2Vycm9yX2NvZGWUXZSME2VuYWJsZWRfZXJyb3JfY29kZXOUj5SME3NjcmlwdHNfYXJlX21vZHVsZXOUiYwLY29uZmlnX2ZpbGWUTowPcXVpY2tzdGFydF9maWxllE6MBWZpbGVzlE6MCHBhY2thZ2VzlE6MB21vZHVsZXOUTowJanVuaXRfeG1slE6MC2luY3JlbWVudGFslIiMCWNhY2hlX2RpcpSMCy5teXB5X2NhY2hllIwMc3FsaXRlX2NhY2hllImMC2RlYnVnX2NhY2hllImMEnNraXBfdmVyc2lvbl9jaGVja5SJjBdza2lwX2NhY2hlX210aW1lX2NoZWNrc5SJjBhmaW5lX2dyYWluZWRfaW5jcmVtZW50YWyUiYwSY2FjaGVfZmluZV9ncmFpbmVklImMFnVzZV9maW5lX2dyYWluZWRfY2FjaGWUiYwPZGVidWdfc2VyaWFsaXpllImMBW15cHljlImMC2luc3BlY3Rpb25zlImMDXByZXNlcnZlX2FzdHOUiYwSaW5jbHVkZV9kb2NzdHJpbmdzlImMB3BsdWdpbnOUXZSMEnBlcl9tb2R1bGVfb3B0aW9uc5R9lIwOdW51c2VkX2NvbmZpZ3OUj5SMCXZlcmJvc2l0eZRLAIwDcGRilImMDnNob3dfdHJhY2ViYWNrlImMEHJhaXNlX2V4Y2VwdGlvbnOUiYwPZHVtcF90eXBlX3N0YXRzlImMFGR1bXBfaW5mZXJlbmNlX3N0YXRzlImMEGR1bXBfYnVpbGRfc3RhdHOUiYwaZW5hYmxlX2luY29tcGxldGVfZmVhdHVyZXOUiYwZZW5hYmxlX2luY29tcGxldGVfZmVhdHVyZZRdlIwMdGltaW5nX3N0YXRzlE6ME2xpbmVfY2hlY2tpbmdfc3RhdHOUTowWc2VtYW50aWNfYW5hbHlzaXNfb25seZSJjBV1c2VfYnVpbHRpbnNfZml4dHVyZXOUiYwLc2hhZG93X2ZpbGWUTowTc2hvd19jb2x1bW5fbnVtYmVyc5SIjA5zaG93X2Vycm9yX2VuZJSIjBBoaWRlX2Vycm9yX2NvZGVzlImMFXNob3dfZXJyb3JfY29kZV9saW5rc5SJjAZwcmV0dHmUiYwKZHVtcF9ncmFwaJSJjAlkdW1wX2RlcHOUiYwMbG9naWNhbF9kZXBzlImME2xvY2FsX3BhcnRpYWxfdHlwZXOUiYwFYmF6ZWyUiYwMZXhwb3J0X3R5cGVzlImMDHBhY2thZ2Vfcm9vdJRdlIwJY2FjaGVfbWFwlH2UjAlmYXN0X2V4aXSUiIwSZmFzdF9tb2R1bGVfbG9va3VwlImMEmFsbG93X2VtcHR5X2JvZGllc5SJjBB0cmFuc2Zvcm1fc291cmNllE6MEnNob3dfYWJzb2x1dGVfcGF0aJSIjA1pbnN0YWxsX3R5cGVzlImMD25vbl9pbnRlcmFjdGl2ZZSJjBVtYW55X2Vycm9yc190aHJlc2hvbGSUSv////+MEm5ld190eXBlX2luZmVyZW5jZZSJjBlkaXNhYmxlX3JlY3Vyc2l2ZV9hbGlhc2VzlImMGGVuYWJsZV9yZWN1cnNpdmVfYWxpYXNlc5SJjA9leHBvcnRfcmVmX2luZm+UiYwbZGlzYWJsZV9ieXRlYXJyYXlfcHJvbW90aW9ulImMHGRpc2FibGVfbWVtb3J5dmlld19wcm9tb3Rpb26UiYwYZm9yY2VfdXBwZXJjYXNlX2J1aWx0aW5zlImMEmZvcmNlX3VuaW9uX3N5bnRheJSJdS4=\"
```

## The issue:
Upon closing  VSCode these processes are orphaned forever and wont quit. Opening and closing VSCode 10 times repetitively spawns 20 zombie processes

## Config/Setup:
VSCode 1.86
Mypy Type Checker: v2023.6.0

```json
"mypy-type-checker.interpreter": ["c:\\Programs\\Python390\\python.exe"],
"mypy-type-checker.preferDaemon": true,
"mypy-type-checker.reportingScope": "workspace",
"mypy-type-checker.importStrategy": "useBundled",
    "mypy-type-checker.severity": {
        "error": "Error",
        "note": "Information"
    },
```

P.S. Switching `"mypy-type-checker.preferDaemon": true,` makes no difference to the issue

**Comment by karthiknadig** [comment]
`"mypy-type-checker.preferDaemon": true ` is what causes daemon to be used (i.e, `dmypy` is used instead of `mypy`). Turn this off (or switching to plain `mypy`) for now using `"mypy-type-checker.preferDaemon": false`.

**Comment by Kein** [comment]
> "mypy-type-checker.preferDaemon": true  is what causes daemon to be used 
Yes, it causes daemon to be used, but it does not cause this issue.
The reason I even set `daemon` to `True` is because I saw the same problem in default configuration.  very often closing VSCode would leave 2-3 orphaned python processes. So I went deep into the settings nd tried to change the things, see if anything affects it. It didn't.

I no longer have the extension in question installed, I swapped it for a different mypy one, but that is the gist of it and why I even reported it in a first place.
